regcomp.c

```diff
@@ -1213,6 +1213,32 @@ S_set_regex_pv(pTHX_ RExC_state_t *pRExC_state, REGEXP *Rx)
     SvCUR_set(Rx, p - RX_WRAPPED(Rx));
 }
 
+static void
+release_RExC_state(pTHX_ void *vstate) {
+    RExC_state_t *pRExC_state = (RExC_state_t *)vstate;
+
+    if (RExC_rx_sv)                                                     \
+        SvREFCNT_dec(RExC_rx_sv);                                         \
```

Comment:
I believe this whole block could be replaced by just `SvREFCNT_dec(RExC_rx_sv);` or even `SvREFCNT_dec_set_NULL(RExC_rx_sv);`.